### PR TITLE
fix(dictionnaryOS): fix repaly rule on existing data

### PR DESCRIPTION
**Pull request by stonebuzz**
Bug found after https://github.com/glpi-project/glpi/pull/14897

bad relation declared from ```relation.constant.php```

```_glpi_items_operatingsystems``` instead of ```glpi_items_operatingsystems```

See other relation like 

```php
 'glpi_operatingsystemservicepacks' => [
        'glpi_items_operatingsystems' => 'operatingsystemservicepacks_id',
    ],

    'glpi_operatingsystemversions' => [
        'glpi_items_operatingsystems' => 'operatingsystemversions_id',
    ],

    'glpi_operatingsystemkernels' => [
        'glpi_operatingsystemkernelversions' => 'operatingsystemkernels_id',
    ],

    'glpi_operatingsystemkernelversions' => [
        'glpi_items_operatingsystems' => 'operatingsystemkernelversions_id',
    ],
```

and when an ```OperatingSystem``` is purged GLPI purged ```Item_OperatingSystem``` too.

Behavior not equal to  ```OperatingSystemVersion```,  ```OperatingSystemEdition```,  ```OperatingSystemArchitecture``` etc ...   

(who do nothing)

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by trasher — APPROVED** [review]

**Comment by trasher** [comment]
Seems good, but changes in `CommonDBTM` requires to be tested

**Comment by stonebuzz** [comment]
> Seems good, but changes in `CommonDBTM` requires to be tested

done =)

File: tests/functional/Glpi/Inventory/Assets/OperatingSystem.php [hunk 1]
```diff
@@ -1183,12 +1183,128 @@ public function testInventoryUpdate()
         $list = $os->find();
         $this->integer(count($list))->isIdenticalTo(2);
 
-       //but still only one linked to computer
```
Comment (trasher) [inline tests/functional/Glpi/Inventory/Assets/OperatingSystem.php:1186]: Why remove this one? It seems correct, isn't it?

File: tests/functional/Glpi/Inventory/Assets/OperatingSystem.php [hunk 2]
```diff
@@ -1183,12 +1183,128 @@ public function testInventoryUpdate()
...
+            [
+                'is_active' => 1,
+            ],
+            [
+                "sub_type" =>
+                [
+                    RuleDictionnaryOperatingSystem::class,
+                ]
+            ]
+        );
```
Comment (trasher) [inline tests/functional/Glpi/Inventory/Assets/OperatingSystem.php:1277]:
```suggestion
        $this->boolean(
            $DB->update(
                Rule::getTable(),
                [
                    'is_active' => 1,
                ],
                [
                    "sub_type" =>
                    [
                        RuleDictionnaryOperatingSystem::class,
                    ]
                ]
            )
        )->isTrue();
```

**Review by trasher — APPROVED** [review]

File: src/RuleDictionnaryDropdownCollection.php [hunk 3]
```diff
@@ -56,7 +56,7 @@ public function replayRulesOnExistingDB($offset = 0, $maxtime = 0, $items = [],
             return $this->replayRulesOnExistingDBForModel($offset, $maxtime);
         }
 
-        if (isCommandLine()) {
+        if (isCommandLine() && !defined('TU_USER')) {
```
Comment (cedric-anne) [inline src/RuleDictionnaryDropdownCollection.php:59]: I would prefer to not have to add this.

In tests, you can catch output using `$this->output(...);` syntax.

**Review by trasher — APPROVED** [review]

File: tests/functional/Glpi/Inventory/Assets/OperatingSystem.php [hunk 4]
```diff
@@ -1183,12 +1183,137 @@ public function testInventoryUpdate()
...
+            )
+        )->isTrue();
+
+        //replay rule
+        $this->login('glpi', 'glpi');
+        $start = explode(" ", microtime());
+        $start = $start[0] + $start[1];
+
+        $max = get_cfg_var("max_execution_time");
+        $max = $start + ($max > 0 ? $max / 2.0 : 30.0);
```
Comment (cedric-anne) [inline tests/functional/Glpi/Inventory/Assets/OperatingSystem.php:1288]: You should not change this in a test.
```suggestion
```

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**